### PR TITLE
Invite history modal

### PR DESCRIPTION
**Pull request by buoyad**
Loading+empty states, split between valid and expired. Not tested on mobile. cc @keybase/y2ksquad

### CODE REVIEW

File: shared/teams/team/invites/invite-history.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+              />
+            )}{' '}
+            • {inviteLink.numUses.toLocaleString()} joined
+            {!!inviteLink.lastJoinedUsername && (
+              <Kb.Text type="BodySmall">
+                , most recently{' '}
+                <Kb.ConnectedUsernames
+                  inline={true}
+                  colorFollowing={true}
+                  type="BodySmall"
```
Comment (jakob223) [inline shared/teams/team/invites/invite-history.tsx:156]: Cècile says all colorFollowing usernames should be bold. Perhaps we should make TS enforce that.

File: shared/common-adapters/section-list.d.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export type SectionListRenderItem<ItemT, ExtraT> = (info: {
  * items, and ExtraT is for any extra stuff that is in the section, e.g. a title
  * used by renderSectionHeader
  */
-export type Section<ItemT, ExtraT> = {
+export type Section<ItemT, ExtraT = {}> = {
```
Comment (jakob223) [inline shared/common-adapters/section-list.d.ts:16]: ooh nice

File: shared/teams/team/invites/invite-history.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+  const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
+  useTeamDetailsSubscribe(teamID)
+  const teamDetails = Container.useSelector(s => Constants.getTeamDetails(s, teamID))
+  const loading = teamDetails === Constants.emptyTeamDetails // TODO should be a better way to check this
+  const [showingExpired, setShowingExpired] = React.useState(false)
+
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+  const onClose = () => dispatch(nav.safeNavigateUpPayload())
+  const onGenerate = () => {} // TODO
```
Comment (jakob223) [inline shared/teams/team/invites/invite-history.tsx:31]: route in Daniel's PR is `teamInviteLinksModal`

File: shared/teams/team/invites/invite-history.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+const splitInviteLinks = memoize((inviteLinks: Set<Types.InviteLink>): {
+  active: Array<Types.InviteLink>
+  expired: Array<Types.InviteLink>
+} => ({active: [...inviteLinks].filter(i => !i.expired), expired: [...inviteLinks].filter(i => i.expired)}))
+
+const InviteHistory = (props: Props) => {
+  const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
+  useTeamDetailsSubscribe(teamID)
+  const teamDetails = Container.useSelector(s => Constants.getTeamDetails(s, teamID))
+  const loading = teamDetails === Constants.emptyTeamDetails // TODO should be a better way to check this
```
Comment (jakob223) [inline shared/teams/team/invites/invite-history.tsx:25]:
```
const {loadingTeam, teamDetails} = Common.useTeamDetailsWithSubscribe(teamID)

useTeamDetailsWithSubscribe = (teamID) => {
useTeamDetailsSubscribe(teamID);   
const teamDetails = Container.useSelector(s => Constants.getTeamDetails(s, teamID))
  const loadingTeam = teamDetails === Constants.emptyTeamDetails
return {teamDetails, loadingTeam}
}
```
WDYT?

File: shared/teams/team/invites/invite-history.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+      )}
+    </Kb.Modal>
+  )
+}
+
+const InviteItem = React.memo(
+  ({inviteLink, teamID}: {inviteLink: Types.InviteLink; teamID: Types.TeamID}) => {
+    const dispatch = Container.useDispatch()
+    const yourUsername = Container.useSelector(s => s.config.username)
+    const onExpire = () => dispatch(TeamsGen.createRemovePendingInvite({inviteID: inviteLink.id, teamID}))
```
Comment (jakob223) [inline shared/teams/team/invites/invite-history.tsx:124]: image
onExpire should trigger this waiting state.

Comment (jakob223) [reply shared/teams/team/invites/invite-history.tsx:124]: a progress indicator might be nice too

File: shared/teams/team/invites/invite-history.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+import * as TeamsGen from '../../../actions/teams-gen'
+import {memoize} from '../../../util/memoize'
+import {Section} from '../../../common-adapters/section-list'
+import {formatDurationLong} from '../../../util/timestamp'
+import {useTeamDetailsSubscribe} from '../../subscriber'
+import {ModalTitle} from '../../common'
+
+type Props = Container.RouteProps<{teamID: Types.TeamID}>
+
+const splitInviteLinks = memoize((inviteLinks: Set<Types.InviteLink>): {
```
Comment (jakob223) [inline shared/teams/team/invites/invite-history.tsx:16]: when I click "expire link" it doesn't move to the other section, it just vanishes.

File: shared/teams/team/invites/invite-history.tsx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+        style={Styles.globalStyles.flexOne}
+        centerChildren={true}
+      >
+        {loading ? (
+          <Kb.Box2 direction="vertical" fullWidth={true} gap="tiny" centerChildren={true}>
+            <Kb.ProgressIndicator type="Large" />
+            <Kb.Text type="BodySmall">Loading...</Kb.Text>
+          </Kb.Box2>
+        ) : (
+          <Kb.Box2 direction="vertical" fullWidth={true} gap="tiny" centerChildren={true}>
```
Comment (jakob223) [inline shared/teams/team/invites/invite-history.tsx:57]: This empty state doesn't appear for me.
image

File: shared/teams/team/invites/invite-history.tsx [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+  )
+}
+
+const InviteItem = React.memo(
+  ({inviteLink, teamID}: {inviteLink: Types.InviteLink; teamID: Types.TeamID}) => {
+    const dispatch = Container.useDispatch()
+    const yourUsername = Container.useSelector(s => s.config.username)
+    const onExpire = () => dispatch(TeamsGen.createRemovePendingInvite({inviteID: inviteLink.id, teamID}))
+
+    const duration = formatDurationLong(new Date(), new Date(inviteLink.expirationTime * 1000))
```
Comment (jakob223) [inline shared/teams/team/invites/invite-history.tsx:126]: date-fns has a utility for this. https://github.com/keybase/client/pull/23517/files#diff-99dcdcb4325f258c146cc7e98db3127eR207

File: shared/teams/team/invites/invite-history.tsx (hunk 6 above)
Comment (buoyad) [reply shared/teams/team/invites/invite-history.tsx:16]: known, ticketed separately

File: shared/teams/team/invites/invite-history.tsx (hunk 7 above)
Comment (buoyad) [reply shared/teams/team/invites/invite-history.tsx:57]: I can't repro this, are you possibly out of date?

Comment (buoyad) [reply shared/teams/team/invites/invite-history.tsx:57]: the most recent commit adds the empty state

File: shared/teams/team/invites/invite-history.tsx (hunk 4 above)
Comment (buoyad) [reply shared/teams/team/invites/invite-history.tsx:25]: the object equality part seems fragile. i'll just pull directly from the map for now

**Comment by buoyad** [comment]
@jakob223 thanks! ready for another look

**Pull request merged**